frontend/lib/framework/error-classes-list.cpp

```diff
@@ -553,6 +553,18 @@ void ErrorSuperFromTopLevelModule::write(ErrorWriterBase& wr) const {
   wr.code(mod);
 }
 
+void ErrorIfVarNonClassType::write(ErrorWriterBase& wr) const {
+  auto cond = std::get<const uast::Conditional*>(info);
+  auto qtVar = std::get<types::QualifiedType>(info);
+  auto var = cond->condition()->toVariable();
+  CHPL_ASSERT(var);
+  auto ifKindStr = cond->isExpressionLevel() ? "expression" : "statement";
+  wr.heading(kind_, type_, var, "a variable declared in the condition of "
+                                "an if ", ifKindStr, " must be a class, "
+                                "not ", decayToValue(qtVar), ".");
```

Comment:
"a class, not a value of type int" reads confusingly (it mixes types and values). Perhaps ", but it has type " and then just print the .type() of `qtVar`?